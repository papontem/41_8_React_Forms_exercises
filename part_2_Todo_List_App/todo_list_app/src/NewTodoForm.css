/* NewTodoForm.css */

.NewTodoForm-container {
    margin: 5px;
}

.NewTodoForm-title{
    font-size:calc(10px + 1vmin);
}
.NewTodoForm {
    /* Add additional styling for the form if needed */
}

.NewTodoForm-label {
    margin: 5px;
}

.NewTodoForm-input {
    margin: 5px;
}

.NewTodoForm-btn {
    margin: 5px;
    padding: 8px 12px;
    background-color: #007bff;
    color: #fff;
    border: none;
    cursor: pointer;
}

.NewTodoForm-btn:hover {
    background-color: #0056b3;
}
