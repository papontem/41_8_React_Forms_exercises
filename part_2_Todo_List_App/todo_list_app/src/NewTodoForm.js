import React, { useState } from "react";
/* 
    - Renders a form with one text input for creating a new task.
    - When the form is submitted, it creates a new Todo component.
*/
function NewTodoForm({ addTodo }) {
	// State for form data
	// width, height, and background color for a new Todo.
	const [newTodoFormData, setNewTodoFormData] = useState({
		task: ""
	});

	// Event handler for form input changes
	const handleChange = (e) => {
		const { name, value } = e.target;
		setNewTodoFormData((prevData) => ({
			...prevData,
			[name]: value,
		}));
	};

	// Event handler for form submission
	const handleSubmit = (e) => {
		e.preventDefault();

		console.log("Form submitted:", newTodoFormData);
		addTodo(newTodoFormData);

		// Clear input values after form submission
		setNewTodoFormData({
			task: "",
		});

		// Now create the Todo
	};

	return (
		<div className="NewTodoForm-container">
			<h2 className="NewTodoForm-title">New Todo Form</h2>
			<form className="NewTodoForm" onSubmit={handleSubmit}>
				<label className="NewTodoForm-label">
					task:
					<input
						className="NewTodoForm-input"
						type="text"
						name="task"
						value={newTodoFormData.task}
						onChange={handleChange}
						required
					/>
				</label>
				<br/>

				<button className="NewTodoForm-btn" type="submit">
					Add New Todo!
				</button>
			</form>
		</div>
	);
}

export default NewTodoForm;
