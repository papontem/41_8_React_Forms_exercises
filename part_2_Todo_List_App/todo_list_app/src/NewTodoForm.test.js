import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // For expect assertions
import NewTodoForm from "./NewTodoForm";

test("renders NewTodoForm without crashing", () => {
  render(<NewTodoForm />);

  // Check if the new box form heading is rendered
  expect(screen.getByText('New Todo Form')).toBeInTheDocument();
});

test('NewTodoForm snapshot', () => {
  // Render the component
  const { asFragment } = render(<NewTodoForm />);

  // Take a snapshot of the rendered component
  expect(asFragment()).toMatchSnapshot();
});
