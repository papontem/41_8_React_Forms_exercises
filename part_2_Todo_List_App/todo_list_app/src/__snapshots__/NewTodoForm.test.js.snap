// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewTodoForm-container"
  >
    <h2
      class="NewTodoForm-title"
    >
      New Todo Form
    </h2>
    <form
      class="NewTodoForm"
    >
      <label
        class="NewTodoForm-label"
      >
        task:
        <input
          class="NewTodoForm-input"
          name="task"
          required=""
          type="text"
          value=""
        />
      </label>
      <br />
      <button
        class="NewTodoForm-btn"
        type="submit"
      >
        Add New Todo!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
